### PR TITLE
Refactor test suite: Automated Testing Strategy for Mainnet with ADA-Free Scenarios

### PR DESCRIPTION
**Pull request by kneerose**
## List of changes

- Separate ADA-independent tests into a different file for better organization
- Skipped ADA-spending tests and added Allure descriptions for improved reporting.
- Configured networkId to dynamically switch and use a mainnet wallet for mainnet tests.
- Modified test 1D to validate across different networks instead of being mainnet-specific.
- Implemented a workflow dispatch mechanism to trigger Playwright automated tests on gov.tools.
- Added logic to clean up proposals created during tests to avoid clutter and maintain test integrity.

## Checklist

- [related issue](https://github.com/cardanoapi/govtool-test-reports/issues/8)
- [x] My changes generate no new warnings
- [x] My code follows the [style guidelines](https://github.com/IntersectMBO/govtool/tree/main/docs/style-guides) of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the [changelog](https://github.com/IntersectMBO/govtool/blob/main/CHANGELOG.md)
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: tests/govtool-frontend/playwright/tests/2-delegation/delegation.loggedin.spec.ts [hunk 1]
```diff
@@ -71,3 +74,55 @@ test("2X_2. Should include info button and voting power on the Signal-No-Confide
 
   await expect(dRepDirectoryPage.signalNoConfidenceInfoButton).toBeVisible();
 });
+
+test.describe("DRep dependent tests", () => {
+  let dRepGivenName = "test";
+  let dRepId = "drep1ef7uslcjhjqrn4vv2y39c3yn345gzjsg7yufn76zye3v6fkz23q";
```
Comment (mesudip) [inline tests/govtool-frontend/playwright/tests/2-delegation/delegation.loggedin.spec.ts:80]: shouldn't this be based on network?

File: tests/govtool-frontend/playwright/tests/2-delegation/delegation.spec.ts [hunk 2]
```diff
@@ -122,3 +125,76 @@ test("2K_1. Should filter DReps", async ({ page }) => {
...
+}) => {
+  const dRepDirectoryPage = new DRepDirectoryPage(page);
+  await dRepDirectoryPage.goto();
+
+  const dRepCards = await dRepDirectoryPage.getAllListedDReps();
+  expect(dRepCards.length).toBeGreaterThan(1);
+});
+
+test.describe("DRep dependent tests", () => {
+  let dRepId = "drep1ef7uslcjhjqrn4vv2y39c3yn345gzjsg7yufn76zye3v6fkz23q";
```
Comment (mesudip) [inline tests/govtool-frontend/playwright/tests/2-delegation/delegation.spec.ts:140]: Should be different based on current network.

**Review by mesudip — CHANGES_REQUESTED** [review]

File: tests/govtool-frontend/playwright/tests/2-delegation/delegation.loggedin.spec.ts (hunk 1 above)
Comment (kneerose) [reply tests/govtool-frontend/playwright/tests/2-delegation/delegation.loggedin.spec.ts:80]: @mesudip , This is an example scenario when no active dReps are in the list. The dRep ID and dRep name are being overwritten by the values from fetchFirstActiveDRepDetails. Should I update this behavior based on the network? What happens if dbsync is cleared on the testnet, and that specific dRep is no longer available in the dRep list?

**Review by mesudip — APPROVED** [review]
Looks good

**Pull request merged**